tests/console/commands/test_check.py

```diff
@@ -74,3 +111,51 @@ def test_check_private(
 """
 
     assert tester.io.fetch_output() == expected
+
+
+def test_check_lock_outdated(
+    command_tester_factory: CommandTesterFactory,
+    poetry_with_outdated_lockfile: Poetry,
+    http: type[httpretty.httpretty],
+) -> None:
+    http.disable()
+
+    locker = Locker(
+        lock=poetry_with_outdated_lockfile.pyproject.file.path.parent / "poetry.lock",
+        local_config=poetry_with_outdated_lockfile.locker._local_config,
+    )
+    poetry_with_outdated_lockfile.set_locker(locker)
+
+    tester = command_tester_factory("check", poetry=poetry_with_outdated_lockfile)
+    status_code = tester.execute("--lock")
+    expected = (
+        "Error: poetry.lock is not consistent with pyproject.toml. "
+        "Run `poetry lock [--no-update]` to fix it.\n"
+    )
+
+    assert tester.io.fetch_error() == expected
+
+    # exit with an error
+    assert status_code == 1
+
+
+def test_check_lock_up_to_date(
+    command_tester_factory: CommandTesterFactory,
+    poetry_with_up_to_date_lockfile: Poetry,
+    http: type[httpretty.httpretty],
+) -> None:
+    http.disable()
+
+    locker = Locker(
+        lock=poetry_with_up_to_date_lockfile.pyproject.file.path.parent / "poetry.lock",
+        local_config=poetry_with_up_to_date_lockfile.locker._local_config,
+    )
+    poetry_with_up_to_date_lockfile.set_locker(locker)
+
+    tester = command_tester_factory("check", poetry=poetry_with_up_to_date_lockfile)
+    status_code = tester.execute("--lock")
```

Comment:
Same here (parametrize).

Reply:
Good catch, changed in 275ff17fc269c79d21805b6f5e3c1a1850f8c808